Meanwhile, please fix the issues highlighted in the `ci/circleci`:

```
stable/hlf-ca/values.yaml
  57:3      error    trailing spaces  (trailing-spaces)
  62:15     error    trailing spaces  (trailing-spaces)
  70:89     error    trailing spaces  (trailing-spaces)
  74:20     error    trailing spaces  (trailing-spaces)
  85:86     error    trailing spaces  (trailing-spaces)
  86:128    error    trailing spaces  (trailing-spaces)
  93:15     error    trailing spaces  (trailing-spaces)
  97:121    error    trailing spaces  (trailing-spaces)
  104:8     error    trailing spaces  (trailing-spaces)
```

--- a/stable/hlf-ca/Chart.yaml
+++ b/stable/hlf-ca/Chart.yaml
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: Hyperledger Fabric Certificate Authority chart (these charts are created by AID:Tech and are currently not directly associated with the Hyperledger project)
 name: hlf-ca
-version: 1.1.0
+version: 1.1.1
 appVersion: 1.2.0
 keywords:
   - blockchain

--- a/stable/hlf-ca/README.md
+++ b/stable/hlf-ca/README.md
@@ -81,6 +81,7 @@ The following table lists the configurable parameters of the Hyperledger Fabric 
 | `ingress.path`                     | Ingress path                                     | `/`                                                        |
 | `ingress.hosts`                    | Ingress hostnames                                | `[]`                                                       |
 | `ingress.tls`                      | Ingress TLS configuration                        | `[]`                                                       |
+| `persistence.existingClaim`        | Name of an existing PVC to use for Fabric CA     | `nil`                                            |
 | `persistence.accessMode`           | Use volume as ReadOnly or ReadWrite              | `ReadWriteOnce`                                            |
 | `persistence.annotations`          | Persistent Volume annotations                    | `{}`                                                       |
 | `persistence.size`                 | Size of data volume                              | `1Gi`                                                      |

--- a/stable/hlf-ca/README.md
+++ b/stable/hlf-ca/README.md
@@ -81,7 +81,7 @@ The following table lists the configurable parameters of the Hyperledger Fabric 
 | `ingress.path`                     | Ingress path                                     | `/`                                                        |
 | `ingress.hosts`                    | Ingress hostnames                                | `[]`                                                       |
 | `ingress.tls`                      | Ingress TLS configuration                        | `[]`                                                       |
-| `persistence.existingClaim`        | Name of an existing PVC to use for Fabric CA     | `nil`                                            |
+| `persistence.existingClaim`        | Name of an existing PVC to use for Fabric CA     | `nil`                                                      |
 | `persistence.accessMode`           | Use volume as ReadOnly or ReadWrite              | `ReadWriteOnce`                                            |
 | `persistence.annotations`          | Persistent Volume annotations                    | `{}`                                                       |
 | `persistence.size`                 | Size of data volume                              | `1Gi`                                                      |
@@ -89,8 +89,15 @@ The following table lists the configurable parameters of the Hyperledger Fabric 
 | `adminUsername`                    | Admin Username for CA                            | `admin`                                                    |
 | `adminPassword`                    | Admin Password                                   | Random 24 alphanumeric characters                          |
 | `caName`                           | Name of CA                                       | `org1-ca`                                                  |
-| `db.chart`                         | Name of a Database Chart holding CA data         | `` supports postgresql                                     |
-| `db.database`                      | Name of the actual Database holding the CA data  | `fabric_ca`                                                |
+| `db.ssl`                           | SSL Authentication                               | `disable`                                                  |
+| `postgresql.enabled`               | Deploy a PostgreSQL container holding the CA data | `false`                                                   |
+| `mysql.enabled`                    | Deploy a MySQL container holding the CA data     | `false`                                                    |
+| `externalDatabase.type`            | Database type (either `postgres` or `mysql` )    | `nil`                                                      |
+| `externalDatabase.host`            | Host of the external database                    | `localhost`                                                |
+| `externalDatabase.username`        | Existing username in the external db             | ``                                                         |
+| `externalDatabase.password`        | Password for the above username                  | ``                                                         |
+| `externalDatabase.database`        | Name of the existing database                    | ``                                                         |
+| `externalDatabase.port`            | Database port number                             | ``                                                         |
 | `config.hlfToolsVersion`           | Version of Hyperledger Fabric tools used         | `1.1.0`                                                    |
 | `config.mountTLS`                  | If TLS secrets are generated, do we mount them?  | `false`                                                    |
 | `config.debug`                     | Enable debug logging                             | `true`                                                     |
@@ -109,6 +116,10 @@ The following table lists the configurable parameters of the Hyperledger Fabric 
 | `nodeSelector`                     | Node labels for pod assignment                   | `{}`                                                       |
 | `tolerations`                      | Toleration labels for pod assignment             | `[]`                                                       |
 | `affinity`                         | Affinity settings for pod assignment             | `{}`                                                       |
+
+## Database
+
+TBD
 
 ## Persistence
 

--- a/stable/hlf-ca/requirements.lock
+++ b/stable/hlf-ca/requirements.lock
@@ -1,0 +1,9 @@
+dependencies:
+- name: postgresql
+  repository: https://kubernetes-charts.storage.googleapis.com/
+  version: 1.0.0
+- name: mysql
+  repository: https://kubernetes-charts.storage.googleapis.com/
+  version: 0.10.1
+digest: sha256:23aa42d69def76d3ca97ade89e504fc7aa7632420d479c52341dfa5b024aa9e1
+generated: 2018-10-18T16:12:34.300395535-04:00

--- a/stable/hlf-ca/requirements.yaml
+++ b/stable/hlf-ca/requirements.yaml
@@ -1,0 +1,14 @@
+dependencies:
+- name: postgresql
+  version: x.x.x
+  repository: https://kubernetes-charts.storage.googleapis.com/
+  condition: postgresql.enabled
+  tags:
+    - postgres-database
+
+- name: mysql
+  version: x.x.x
+  repository: https://kubernetes-charts.storage.googleapis.com/
+  condition: mysql.enabled
+  tags:
+    - mysql-database

--- a/stable/hlf-ca/templates/_helpers.tpl
+++ b/stable/hlf-ca/templates/_helpers.tpl
@@ -43,3 +43,17 @@ heritage: {{ .Release.Service | quote }}
 release: {{ .Release.Name | quote }}
 chart: {{ include "hlf-ca.chart" . }}
 {{- end -}}
+
+{{/*
+Generate postgres chart secret name
+*/}}
+{{- define "postgresql.secretName" -}}
+{{- printf "%s-%s" .Release.Name "postgresql" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Generate mysql chart secret name
+*/}}
+{{- define "mysql.secretName" -}}
+{{- printf "%s-%s" .Release.Name "mysql" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}

--- a/stable/hlf-ca/templates/configmap--db.yaml
+++ b/stable/hlf-ca/templates/configmap--db.yaml
@@ -5,14 +5,28 @@ metadata:
   labels:
 {{ include "labels.standard" . | indent 4 }}
 data:
-  # TODO: Enable multiple types of database, not only PostgreSQL
-  DB_TYPE: postgres
-  DB_USERNAME: postgres
-  DB_DATABASE: {{ .Values.db.database | quote }}
-  {{- if .Values.db.chart }}
-  DB_HOST: {{ .Values.db.chart }}.{{ .Release.Namespace }}.svc.cluster.local
-  {{- else }}
-  DB_HOST: localhost
+  DB_SSL: {{ .Values.db.ssl | quote }}
+  {{- if .Values.postgresql.enabled }}
+  # PostgreSQL Database
+  DB_TYPE: "postgres"
+  DB_HOST: {{ template "postgresql.fullname" . }}
+  DB_PORT: {{ .Values.postgresql.port | quote }}
+  DB_USERNAME: {{ .Values.postgresql.postgresUser | quote }}
+  DB_DATABASE: {{ .Values.postgresql.postgresDatabase  | quote }}
   {{- end }}
-  DB_PORT: "5432"
-  DB_SSL: disable
+  {{- if .Values.mysql.enabled }}
+  # MySQL Database
+  DB_TYPE: "mysql"
+  DB_HOST: {{ template "mysql.fullname" . }}
+  DB_PORT: "3306"
+  DB_USERNAME: {{ .Values.mysql.mysqlUser | quote }}
+  DB_DATABASE: {{ .Values.mysql.mysqlDatabase  | quote }}
+  {{- end }}
+  {{- if .Values.externalDatabase.type }}
+  # External Database
+  DB_TYPE: {{ .Values.externalDatabase.type | quote }}
+  DB_HOST: {{ .Values.externalDatabase.host | quote }}
+  DB_PORT: {{ .Values.externalDatabase.port | quote }}
+  DB_USERNAME: {{ .Values.externalDatabase.username | quote }}
+  DB_DATABASE: {{ .Values.externalDatabase.database  | quote }}
+  {{- end }}

--- a/stable/hlf-ca/templates/configmap--db.yaml
+++ b/stable/hlf-ca/templates/configmap--db.yaml
@@ -26,7 +26,12 @@ data:
   # External Database
   DB_TYPE: {{ .Values.externalDatabase.type | quote }}
   DB_HOST: {{ .Values.externalDatabase.host | quote }}
-  DB_PORT: {{ .Values.externalDatabase.port | quote }}
+  {{- if eq .Values.externalDatabase.type "mysql" }}
+  DB_PORT: {{ .Values.externalDatabase.port | default "3306" | quote }}
+  {{- end }}
+  {{- if eq .Values.externalDatabase.type "postgres" }}
+  DB_PORT: {{ .Values.externalDatabase.port | default "5432" | quote }}
+  {{- end }}
   DB_USERNAME: {{ .Values.externalDatabase.username | quote }}
   DB_DATABASE: {{ .Values.externalDatabase.database  | quote }}
   {{- end }}

--- a/stable/hlf-ca/templates/deployment.yaml
+++ b/stable/hlf-ca/templates/deployment.yaml
@@ -40,19 +40,21 @@ spec:
             - sh
             - -c
             - |
-              while [ -z ${DB_PASSWORD+x} ];
-              do
-                echo "\033[0;31m DB_PASSWORD environmental variable must contain a value (check that you specified 'db.chart' in values.yaml) \033[0m"
-                sleep 60
-              done
+              if [ -n "$DB_TYPE" ]; then
+                while [ -z ${DB_PASSWORD+x} ];
+                do
+                  echo "\033[0;31m DB_PASSWORD environmental variable must contain a value \033[0m"
+                  sleep 60
+                done
+                export DB_DATASOURCE="host=${DB_HOST} port=${DB_PORT} user=${DB_USERNAME} password=${DB_PASSWORD} dbname=${DB_DATABASE} sslmode=${DB_SSL}"
+              fi
 
-              export DB_DATASOURCE="host=${DB_HOST} port=${DB_PORT} user=${DB_USERNAME} password=${DB_PASSWORD} dbname=${DB_DATABASE} sslmode=${DB_SSL}"
               mkdir -p $FABRIC_CA_HOME
               cp /var/hyperledger/ca_config/ca.yaml $FABRIC_CA_HOME/fabric-ca-server-config.yaml
               sed -i "s/<<<ADMIN>>>/${CA_ADMIN}/g" ${FABRIC_CA_HOME}/fabric-ca-server-config.yaml
               sed -i "s/<<<ADMINPW>>>/${CA_PASSWORD}/g" $FABRIC_CA_HOME/fabric-ca-server-config.yaml
-              sed -i "s/<<<DB_TYPE>>>/${DB_TYPE}/g" $FABRIC_CA_HOME/fabric-ca-server-config.yaml
-              sed -i "s/<<<DB_DATASOURCE>>>/${DB_DATASOURCE}/g" $FABRIC_CA_HOME/fabric-ca-server-config.yaml
+              sed -i "s/<<<DB_TYPE>>>/${DB_TYPE:-"sqlite3"}/g" $FABRIC_CA_HOME/fabric-ca-server-config.yaml
+              sed -i "s/<<<DB_DATASOURCE>>>/${DB_DATASOURCE:-"fabric-ca-server.db"}/g" $FABRIC_CA_HOME/fabric-ca-server-config.yaml
 
               {{- if .Values.config.intermediate }}
               echo ">\033[0;33m Adding Parent CA (typically a Root CA) values to $FABRIC_CA_HOME/fabric-ca-server-config.yaml \033[0m"
@@ -64,21 +66,38 @@ spec:
               fabric-ca-server start
           envFrom:
             - secretRef:
-                name: {{ include "hlf-ca.fullname" . }}
+                name: {{ include "hlf-ca.fullname" . }}--ca
             - configMapRef:
                 name: {{ include "hlf-ca.fullname" . }}--db
             - configMapRef:
                 name: {{ include "hlf-ca.fullname" . }}--ca
           env:
-            # Database environmental variable
-            {{- if .Values.db.chart }}
+            {{- if .Values.postgresql.enabled}}
+            # PostgreSQL Database
             - name: DB_PASSWORD
               valueFrom:
                 secretKeyRef:
-                  name: {{ .Values.db.chart | quote }}
+                  name: {{ template "postgresql.secretName" . }}
                   key: postgres-password
             {{- end }}
+            {{- if .Values.mysql.enabled }}
+            # MySQL Database
+            - name: DB_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ template "mysql.secretName" . }}
+                  key: mysql-password
+            {{- end }}
+            {{- if .Values.externalDatabase.type }}
+            # External Database
+            - name: DB_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "hlf-ca.fullname" . }}--db
+                  key: db-password
+            {{- end }}            
             {{- if .Values.config.intermediate }}
+            # Intermediate CA
             - name: PARENT_CA_ADMIN
               valueFrom:
                 secretKeyRef:

--- a/stable/hlf-ca/templates/secret--ca.yaml
+++ b/stable/hlf-ca/templates/secret--ca.yaml
@@ -1,7 +1,7 @@
 apiVersion: v1
 kind: Secret
 metadata:
-  name: {{ include "hlf-ca.fullname" . }}
+  name: {{ include "hlf-ca.fullname" . }}--ca
   labels:
 {{ include "labels.standard" . | indent 4 }}
 type: Opaque

--- a/stable/hlf-ca/templates/secret--db.yaml
+++ b/stable/hlf-ca/templates/secret--db.yaml
@@ -1,0 +1,11 @@
+{{- if .Values.externalDatabase.type }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "hlf-ca.fullname" . }}--db
+  labels:
+{{ include "labels.standard" . | indent 4 }}
+type: Opaque
+data:
+  db-password: {{ .Values.externalDatabase.password | b64enc | quote }}
+{{- end }}

--- a/stable/hlf-ca/values.yaml
+++ b/stable/hlf-ca/values.yaml
@@ -49,10 +49,53 @@ adminUsername: admin
 ## Name of CA, stored in FABRIC_CA_SERVER_CA_NAME (and the configMap "--config", if deploying a Root CA)
 caName: org1-ca
 
-## Settings used in configMap "--db"
+##
+## Fabric CA Database
+## To setup a database, set the param
+##################################
+## 
 db:
-  # chart: postgres-ca-postgresql
-  database: fabric_ca
+  # SSL authentication
+  # Supported values are: "disable", "require", "verify-ca" and "verify-full".
+  # https://github.com/hyperledger/fabric-ca/blob/master/docs/source/users-guide.rst
+  ssl: disable 
+
+##
+## PostgreSQL  chart configuration
+##
+## https://github.com/helm/charts/tree/master/stable/postgresql
+##
+postgresql:
+  ## Whether to deploy a postgres server to satisfy the Fabric CA database requirements. 
+  # To use an external database set this to false and configure the externalDatabase parameters 
+  enabled: false
+  port: 5432
+  postgresDatabase: fabric_ca
+  postgresUser: fabric_ca
+
+##
+## MySQL chart configuration
+##
+## https://github.com/helm/charts/tree/master/stable/mysql
+##
+mysql:
+  ## Whether to deploy a mysql server to satisfy the Fabric CA database requirements. 
+  # To use an external database set this to false and configure the externalDatabase parameters  
+  enabled: false
+  mysqlDatabase: fabric_ca
+  mysqlUser: fabric_ca
+
+##
+## External Database configuration (MySQL or PostgresSQL install outside Kubernetes)
+##
+externalDatabase:
+  # Either postgres or mysql
+  type: 
+  host: ""
+  port: ""
+  database: ""
+  username: ""
+  password: ""
 
 ## Settings used in configMap "--config"
 config:

